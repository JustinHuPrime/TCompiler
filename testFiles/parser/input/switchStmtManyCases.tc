module foo;

int bar(int i) {
  switch (i) {
    case 0: return -1;
    case 1:
    case -1: { return 0; }
    default: return 1;
  }
}