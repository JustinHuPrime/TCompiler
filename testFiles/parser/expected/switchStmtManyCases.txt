testFiles/parser/switchStmtManyCases.tc (code):
FILE(1, 1, STAB(ENTRY(bar, FUNCTION(testFiles/parser/switchStmtManyCases.tc, 3, 1, int(int)))), MODULE(1, 1, ID(1, 8, foo, REFERENCES())), FUNDEFN(3, 1, KEYWORDTYPE(3, 1, int), ID(3, 5, bar, REFERENCES(testFiles/parser/switchStmtManyCases.tc, 3, 1)), KEYWORDTYPE(3, 9, int), ID(3, 13, i, REFERENCES()), STAB(ENTRY(i, VARIABLE(testFiles/parser/switchStmtManyCases.tc, 3, 9, int))), COMPOUNDSTMT(3, 16, STAB(), SWITCHSTMT(4, 3, ID(4, 11, i, REFERENCES(testFiles/parser/switchStmtManyCases.tc, 3, 9)), SWITCHCASE(5, 5, LITERAL(5, 10, UBYTE(0)), RETURNSTMT(5, 13, LITERAL(5, 20, BYTE(-1))), STAB()), SWITCHCASE(6, 5, LITERAL(6, 10, UBYTE(1)), LITERAL(7, 10, BYTE(-1)), COMPOUNDSTMT(7, 14, STAB(), RETURNSTMT(7, 16, LITERAL(7, 23, UBYTE(0)))), STAB()), SWITCHDEFAULT(8, 5, RETURNSTMT(8, 14, LITERAL(8, 21, UBYTE(1))), STAB())))))
